internal/factsengine/gatherers/verifypassword_test.go

```diff
@@ -0,0 +1,90 @@
+package gatherers
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	mocks "github.com/trento-project/agent/internal/factsengine/gatherers/mocks"
+)
+
+type PasswordTestSuite struct {
+	suite.Suite
+}
+
+func TestPasswordTestSuite(t *testing.T) {
+	suite.Run(t, new(PasswordTestSuite))
+}
+
+func (suite *PasswordTestSuite) TestPasswordGatherEqual() {
+	mockExecutor := new(mocks.CommandExecutor)
+
+	shadow := []byte("hacluster:$6$WFEgPAefduOyvLCN$MprO90En7b/" +
+		"cP8uJJpHzJ7ufTPjYuWoVF4s.3MUdOR9iwcO.6E3uCHX1waqypjey458NKGE9O7lnWpV/" +
+		"qd2tg1:19029::::::")
+
+	mockExecutor.On("Exec", "getent", "shadow", haclusterUser).Return(
+		shadow, nil)
+
+	p := &VerifyPasswordGatherer{
+		executor: mockExecutor,
+	}
+
+	factRequests := []FactRequest{
+		{
+			Name:     "hacluster",
+			Gatherer: "password",
+			Argument: "linux",
+			CheckID:  "check1",
+		},
+	}
```

Comment:
Here's how the DSL would look like, right?
```
facts:
    - name: <whatever_name>
      gatherer: password
      argument: linux
```
I am wondering how I would read such a piece of DSL :thinking:

If I understand correctly what you say here
> The aim of this one is to basically check that the hacluster user password has been changes from the default one
>

The gathered fact is _hacluster user password has been changed from the default one_ and I understand that being the reason why the result is a boolean.
And the **default password** you mentioned is the provided argument ("linux" in the example)

Initially I was a bit confused by the checking nuance this gatherer has by password equality :sweat_smile:, yet I realize that the fact of interest is the knowledge that the password has changed (yes/no), not the current value, and that requires internally to do the checking.
I believe then that returned bool will be used in check evaluation itself.

So, just to make it closer to a real scenario, a user would write this piece of DSL
```
facts:
    - name: hacluster_password_changed
      gatherer: password
      argument: linux
```
It works, yet I still feel a bit confused.
Maybe `password` is too a generic name for this gatherer, considering we are assuming the user `hacluster`.
And, since we are hardcoding the user, what stops us from doing the same for the default password?
I mean...hardcode/config/whatever

Imagine we have this DSL
```
facts:
    - name: my_hacluster_pass_has_changed
      gatherer: hacluster_password_changed
```
By finding a more specific name for the gatherer in relation to what it is actually doing and encapsulating what is encapsulable, I believe we could have a clearer and more specific definition, also simpler to read imho :sweat_smile:

Reply:
The gatherer name is `verify_password`. 
So it would look like:
```
facts:
    - name: hacluster_password_changed
      gatherer: verify_password
      argument: linux
```

I was waiting to have this extra arguments option to have the option to get the user from the dsl itself. I don't know how useful it would be to check other users password. But I thought that having it this way would make it at least more usable.

Anyway, we can certainly hardcode the password and change the gatherer name.
But I still think that having it generic will be beneficial. Once we have the option to send the user, we could create more checks, and even our users could test if their own system users don't have default passwords, etc

Reply:
> I don't know how useful it would be to check other users password.
>
This is exactly it :smile: 
this sentence makes me think that the password changed check is very specific for a given user (`hacluster`), at least for now.

However, I understand the abstraction you're looking for, something like this I believe, which I find reasonable
```
facts:
    - name: hacluster_password_changed
      gatherer: verify_password_changed # maybe this clarifies more
      arguments: 
          user: hacluster
          password: linux
```
and I understand that the missing possibility to provide multiple arguments is a limitation.

There is another way forward.

I believe @rtorrero had a similar issue and the solution was providing only one argument that follows a known pattern.
Something like `username:password`

What do you think about that?

Reply:
Ok, I will change to your suggestion.